File: templates/problems/01_arrays_hashing/09_longest_consecutive/longest_consecutive_test.go
package longest_consecutive

import "testing"

func TestLongestConsecutive(t *testing.T) {
	tests := []struct {
		name string
		nums []int
		want int
	}{
		{
			name: "basic case",
			nums: []int{100, 4, 200, 1, 3, 2},
			want: 4,
		},
		{
			name: "longer sequence",
			nums: []int{0, 3, 7, 2, 5, 8, 4, 6, 0, 1},
			want: 9,
		},
		{
			name: "empty array",
			nums: []int{},
			want: 0,
		},
		{
			name: "single element",
			nums: []int{1},
			want: 1,
		},
		{
			name: "no consecutive",
			nums: []int{10, 20, 30},
			want: 1,
		},
		{
			name: "duplicates in sequence",
			nums: []int{1, 2, 2, 3, 3, 4},
			want: 4,
		},
		{
			name: "negative numbers",
			nums: []int{-3, -2, -1, 0, 1},
			want: 5,
		},
		{
			name: "all same elements",
			nums: []int{5, 5, 5, 5},
			want: 1,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := longestConsecutive(tt.nums)
			if got != tt.want {
				t.Errorf("longestConsecutive(%v) = %d, want %d", tt.nums, got, tt.want)
			}
		})
	}
}

File: templates/problems/01_arrays_hashing/09_longest_consecutive/typescript/longestConsecutive.test.ts
import { describe, it, expect } from 'vitest';
import { longestConsecutive } from './longestConsecutive';

describe('LongestConsecutive', () => {
    it('basic case', () => {
        expect(longestConsecutive([100, 4, 200, 1, 3, 2])).toBe(4);
    });

    it('longer sequence', () => {
        expect(longestConsecutive([0, 3, 7, 2, 5, 8, 4, 6, 0, 1])).toBe(9);
    });

    it('empty array', () => {
        expect(longestConsecutive([])).toBe(0);
    });

    it('single element', () => {
        expect(longestConsecutive([1])).toBe(1);
    });

    it('no consecutive', () => {
        expect(longestConsecutive([10, 20, 30])).toBe(1);
    });

    it('duplicates in sequence', () => {
        expect(longestConsecutive([1, 2, 2, 3, 3, 4])).toBe(4);
    });

    it('negative numbers', () => {
        expect(longestConsecutive([-3, -2, -1, 0, 1])).toBe(5);
    });

    it('all same elements', () => {
        expect(longestConsecutive([5, 5, 5, 5])).toBe(1);
    });
});
